src/phoenix/server/api/evaluators.py

```diff
@@ -71,6 +246,75 @@ def get_builtin_evaluator_by_id(evaluator_id: int) -> Optional[type[BuiltInEvalu
     return _BUILTIN_EVALUATORS_BY_ID.get(evaluator_id)
 
 
+async def get_llm_evaluators(
+    evaluator_ids: list[GlobalID],
+    session: AsyncSession,
+    llm_client: PlaygroundStreamingClient,
+) -> list[LLMEvaluator]:
+    from phoenix.server.api.types.Evaluator import DatasetLLMEvaluator
+
+    if not evaluator_ids:
+        return []
+
+    llm_evaluator_db_to_node_id: dict[int, GlobalID] = {}
+    for evaluator_id in evaluator_ids:
+        type_name, db_id = from_global_id(evaluator_id)
+        if type_name == DatasetLLMEvaluator.__name__:
+            llm_evaluator_db_to_node_id[db_id] = evaluator_id
+
+    if not llm_evaluator_db_to_node_id:
+        return []
+
+    llm_evaluator_orms: list[models.LLMEvaluator] = list(
+        await session.scalars(
+            select(models.LLMEvaluator).where(
+                models.LLMEvaluator.id.in_(llm_evaluator_db_to_node_id.keys())
+            )
+        )
+    )
+
+    if len(llm_evaluator_orms) < len(llm_evaluator_db_to_node_id):
+        missing_db_ids = set(llm_evaluator_db_to_node_id.keys()) - set(
+            evaluator.id for evaluator in llm_evaluator_orms
+        )
+        missing_node_ids = [llm_evaluator_db_to_node_id[db_id] for db_id in missing_db_ids]
+        raise NotFound(
+            f"Could not find all LLM evaluators with IDs {', '.join(map(_quote, missing_node_ids))}"
+        )
+
+    llm_evaluators: list[LLMEvaluator] = []
+    for llm_evaluator_orm in llm_evaluator_orms:
+        prompt_id = llm_evaluator_orm.prompt_id
+        prompt_version_tag_id = llm_evaluator_orm.prompt_version_tag_id
+        if prompt_version_tag_id is not None:
+            # Get the tagged version
+            prompt_version_query = (
+                select(models.PromptVersion)
+                .join(models.PromptVersionTag)
+                .where(models.PromptVersionTag.id == prompt_version_tag_id)
+            )
+        else:
+            # Get the latest version
+            prompt_version_query = (
```

Comment:
this logic shows up in several places we should probably make sure the way we associate a prompt with an evaluator is abstracted away somewhere (this can be done later)